rules/no-substr.js

```diff
@@ -7,10 +7,9 @@
 module.exports = function (context) {
   return {
     MemberExpression: function (node) {
-      var objectType = node.object.type;
       var propertyName = node.property.name;
 
-      if (objectType === 'Literal' && propertyName === 'substr') {
+      if (propertyName === 'substr') {
```

Comment:
Maybe this would be better instead?

```suggestion
      if (['Identifier', 'Literal'].includes(node.object.type) && propertyName === 'substr') {
```